### PR TITLE
openstack-crowbar: collect supportconfig files on failure (SOC-8460)

### PR DESCRIPTION
**Pull request by stefannica**
Implement an ansible playbook and role to generate and collect
supportconfig files from all cloud nodes.
Collect supportconfig files as Jenkins artifacts by default in all
periodic and gating Crowbar jobs in case of failure.

### CODE REVIEW

**Comment by stefannica** [comment]
Example job run here: https://ci.suse.de/blue/organizations/jenkins/cloud-crowbar9-job-gate-ha-deploy-x86_64/detail/cloud-crowbar9-job-gate-ha-deploy-x86_64/11/pipeline

**Comment by stefannica** [comment]
> Example job run here: https://ci.suse.de/blue/organizations/jenkins/cloud-crowbar9-job-gate-ha-deploy-x86_64/detail/cloud-crowbar9-job-gate-ha-deploy-x86_64/11/pipeline

Previous failed for unrelated reasons. New run here: https://ci.suse.de/blue/organizations/jenkins/cloud-crowbar9-job-gate-ha-deploy-x86_64/detail/cloud-crowbar9-job-gate-ha-deploy-x86_64/13/pipeline

**Comment by stefannica** [comment]
Found a bug. Re-testing with: https://ci.suse.de/blue/organizations/jenkins/cloud-crowbar9-job-gate-ha-deploy-x86_64/detail/cloud-crowbar9-job-gate-ha-deploy-x86_64/15/pipeline

**Comment by stefannica** [comment]
Found another bug. Re-testing with: https://ci.suse.de/blue/organizations/jenkins/cloud-crowbar9-job-gate-ha-deploy-x86_64/detail/cloud-crowbar9-job-gate-ha-deploy-x86_64/17/pipeline

File: scripts/jenkins/cloud/ansible/roles/supportconfig/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  loop: "{{ supportconfig_pkgs }}"
+
+- name: Generate supportconfig
+  shell: |
+    supportconfig -B {{ supportconfig_filename }}
+  args:
+    creates: "/var/log/nts_{{ supportconfig_filename }}.tbz"
+  failed_when: false
+
+- name: Collect supportconfig
```
Comment (flaviodsr) [inline scripts/jenkins/cloud/ansible/roles/supportconfig/tasks/main.yml]: Couldn't you reuse the `jenkins_artifacts` role on the playbook? it does exactly the same thing plus ensures that `jenkins_artifacts_dir` exists.

File: scripts/jenkins/cloud/ansible/collect-supportconfig.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (flaviodsr) [inline scripts/jenkins/cloud/ansible/collect-supportconfig.yml]: could you add the license boilerplate here?

File: scripts/jenkins/cloud/ansible/roles/supportconfig/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (flaviodsr) [inline scripts/jenkins/cloud/ansible/roles/supportconfig/defaults/main.yml]: same here

File: scripts/jenkins/cloud/ansible/roles/supportconfig/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+---
```
Comment (flaviodsr) [inline scripts/jenkins/cloud/ansible/roles/supportconfig/defaults/main.yml]: same here

File: scripts/jenkins/cloud/ansible/roles/supportconfig/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (flaviodsr) [inline scripts/jenkins/cloud/ansible/roles/supportconfig/tasks/main.yml]: same here
also, in some files you add a blank line before `---` and others you don't, could you keep the same for all files?

File: scripts/jenkins/cloud/ansible/collect-supportconfig.yml (hunk 2 above)
Comment (stefannica) [reply scripts/jenkins/cloud/ansible/collect-supportconfig.yml]: done

File: scripts/jenkins/cloud/ansible/roles/supportconfig/tasks/main.yml (hunk 1 above)
Comment (stefannica) [reply scripts/jenkins/cloud/ansible/roles/supportconfig/tasks/main.yml]: That's a good point. Let me try that.

**Comment by stefannica** [comment]
Re-testing changes: https://ci.suse.de/blue/organizations/jenkins/cloud-crowbar9-job-gate-ha-deploy-x86_64/detail/cloud-crowbar9-job-gate-ha-deploy-x86_64/18/pipeline

**Pull request merged**